flavors/common.go

```diff
@@ -57,7 +57,7 @@ func getAWSDataProvider(ctx context.Context, log *logp.Logger, cfg config.Config
 	}
 
 	identityClient := awslib.GetIdentityClient(awsConfig)
-	iamProvider := iam.NewIAMProvider(log, awsConfig)
+	iamProvider := iam.NewIAMProvider(log, awsConfig, nil)
```

Comment:
Not super happy with allowing `nil` here but I don't have a better way right now.

`getAWSDataProvider` doesn't need to use the access accessanalyzer API so having to add the multi-region boilerplate is wrong. The one real alternative is having a whole new fetcher just for access analyzers which seems like a worse solution.

Reply:
I agree, there are better ways in golang to construct instances and we need to handle that soon